src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+    // - isLeftButtonPressed: true iff the left mouse button was pressed during this event.
+    void ControlInteractivity::_mouseScrollHandler(const double mouseDelta,
+                                                   const til::point terminalPosition,
+                                                   const bool isLeftButtonPressed)
+    {
+        const auto currentOffset = _core->ScrollOffset();
+
+        // negative = down, positive = up
+        // However, for us, the signs are flipped.
+        // With one of the precision mice, one click is always a multiple of 120 (WHEEL_DELTA),
+        // but the "smooth scrolling" mode results in non-int values
+        const auto rowDelta = mouseDelta / (-1.0 * WHEEL_DELTA);
+
+        // WHEEL_PAGESCROLL is a Win32 constant that represents the "scroll one page
+        // at a time" setting. If we ignore it, we will scroll a truly absurd number
+        // of rows.
+        const auto rowsToScroll{ _rowsToScroll == WHEEL_PAGESCROLL ? _core->ViewHeight() : _rowsToScroll };
+        double newValue = (rowsToScroll * rowDelta) + (currentOffset);
+
+        // Update the Core's viewport position, and raise a
+        // ScrollPositionChanged event to update the scrollbar
+        _updateScrollbar(::base::saturated_cast<int>(newValue));
+
+        if (isLeftButtonPressed)
+        {
+            // If user is mouse selecting and scrolls, they then point at new
+            // character. Make sure selection reflects that immediately.
+            SetEndSelectionPoint(terminalPosition);
+        }
+    }
+
+    // Method Description:
+    // - Update the scroll position in such a way that should update the
+    //   scrollbar. For example, when scrolling the buffer with the mouse or
+    //   touch input. This will both update the Core's Terminal's buffer
+    //   location, then also raise our own ScrollPositionChanged event.
+    //   UserScrollViewport _won't_ raise the core's ScrollPositionChanged
+    //   event, because it's assumed that's already being called from a context
+    //   that knows about the change to the scrollbar. So we need to raise the
+    //   event on our own.
+    // - The hosting control should make sure to listen to our own
+    //   ScrollPositionChanged event and use that as an opportunity to update
+    //   the location of the scrollbar.
+    // Arguments:
+    // - newValue: The new top of the viewport
+    // Return Value:
+    // - <none>
+    void ControlInteractivity::_updateScrollbar(const int newValue)
+    {
+        _core->UserScrollViewport(newValue);
+
+        // _core->ScrollOffset() is new set to newValue
+        auto scrollArgs = winrt::make_self<ScrollPositionChangedArgs>(_core->ScrollOffset(),
+                                                                      _core->ViewHeight(),
+                                                                      _core->BufferHeight());
+        _ScrollPositionChangedHandlers(*this, *scrollArgs);

```

Comment:
either `winrt::make` it directly in the handler, or do a `std::move` (unsure which is better)